php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+      f.traitDeclaration(
+        b.token(TRAIT),
+        IDENTIFIER(),
+        b.token(LCURLYBRACE),
+        b.zeroOrMore(CLASS_MEMBER()),
+        b.token(RCURLYBRACE)));
+  }
+
+  public ClassDeclarationTree INTERFACE_DECLARATION() {
+    return b.<ClassDeclarationTree>nonterminal(PHPLexicalGrammar.INTERFACE_DECLARATION).is(
+      f.interfaceDeclaration(
+        b.token(INTERFACE),
+        IDENTIFIER(),
+        b.optional(f.newTuple26(b.token(EXTENDS), INTERFACE_LIST())),
+        b.token(LCURLYBRACE),
+        b.zeroOrMore(CLASS_MEMBER()),
+        b.token(RCURLYBRACE)));
+  }
+
+
+  public ClassMemberTree CLASS_MEMBER() {
+    return b.<ClassMemberTree>nonterminal(PHPLexicalGrammar.CLASS_MEMBER).is(
+      b.firstOf(
+        METHOD_DECLARATION(),
+        CLASS_VARIABLE_DECLARATION(),
+        CLASS_CONSTANT_DECLARATION(),
+        USE_TRAIT_DECLARATION()));
+  }
+
+  public ClassPropertyDeclarationTree CLASS_CONSTANT_DECLARATION() {
+    return b.<ClassPropertyDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_CONSTANT_DECLARATION).is(
+      f.classConstantDeclaration(
+        b.token(PHPKeyword.CONST),
+        MEMBER_CONST_DECLARATION(),
+        b.zeroOrMore(f.newTuple97(b.token(COMMA), MEMBER_CONST_DECLARATION())),
+        EOS()));
+  }
+
+  public ConstantDeclarationTree CONSTANT_DECLARATION() {
+    return b.<ConstantDeclarationTree>nonterminal(PHPLexicalGrammar.CONSTANT_DECLARATION).is(
+      f.constantDeclaration(
+        b.token(PHPKeyword.CONST),
+        CONST_VAR(),
+        b.zeroOrMore(f.newTuple28(b.token(COMMA), CONST_VAR())),
+        EOS()));
+  }
+
+  public ClassPropertyDeclarationTree CLASS_VARIABLE_DECLARATION() {
+    return b.<ClassPropertyDeclarationTree>nonterminal(PHPLexicalGrammar.CLASS_VARIABLE_DECLARATION).is(
+      f.classVariableDeclaration(
+        b.firstOf(
+          f.singleToken(b.token(PHPKeyword.VAR)),
+          b.oneOrMore(MEMBER_MODIFIER())),
+        VARIABLE_DECLARATION(),
+        b.zeroOrMore(f.newTuple95(b.token(COMMA), VARIABLE_DECLARATION())),
+        EOS()));
+  }
+
+  public SyntaxToken MEMBER_MODIFIER() {
+    return b.<SyntaxToken>nonterminal(PHPLexicalGrammar.MEMBER_MODIFIER).is(
+      b.firstOf(
+        b.token(PHPKeyword.PUBLIC),
+        b.token(PHPKeyword.PROTECTED),
+        b.token(PHPKeyword.PRIVATE),
+        b.token(PHPKeyword.STATIC),
+        b.token(PHPKeyword.ABSTRACT),
+        b.token(PHPKeyword.FINAL)));
+  }
+
+  public MethodDeclarationTree METHOD_DECLARATION() {
+    return b.<MethodDeclarationTree>nonterminal(PHPLexicalGrammar.METHOD_DECLARATION).is(
+      f.methodDeclaration(
+        b.zeroOrMore(MEMBER_MODIFIER()),
+        b.token(PHPKeyword.FUNCTION),
+        b.optional(b.token(PHPPunctuator.AMPERSAND)),
+        IDENTIFIER(),
+        PARAMETER_LIST(),
+        b.firstOf(
+          EOS(),
+          BLOCK())));
+  }
+
+  public FunctionDeclarationTree FUNCTION_DECLARATION() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)